pxf/pxf-jdbc/src/main/java/org/apache/hawq/pxf/plugins/jdbc/JdbcAccessor.java

```diff
@@ -0,0 +1,469 @@
+                        statementWrite.setNull(i, Types.INTEGER);
+                    }
+                    else {
+                        statementWrite.setShort(i, (short)field.val);
+                    }
+                    break;
+                case REAL:
+                    if (field.val == null) {
+                        statementWrite.setNull(i, Types.FLOAT);
+                    }
+                    else {
+                        statementWrite.setFloat(i, (float)field.val);
+                    }
+                    break;
+                case FLOAT8:
+                    if (field.val == null) {
+                        statementWrite.setNull(i, Types.DOUBLE);
+                    }
+                    else {
+                        statementWrite.setDouble(i, (double)field.val);
+                    }
+                    break;
+                case BOOLEAN:
+                    if (field.val == null) {
+                        statementWrite.setNull(i, Types.BOOLEAN);
+                    }
+                    else {
+                        statementWrite.setBoolean(i, (boolean)field.val);
+                    }
+                    break;
+                case NUMERIC:
+                    if (field.val == null) {
+                        statementWrite.setNull(i, Types.NUMERIC);
+                    }
+                    else {
+                        statementWrite.setBigDecimal(i, (BigDecimal)field.val);
+                    }
+                    break;
+                case VARCHAR:
+                case BPCHAR:
+                case TEXT:
+                    if (field.val == null) {
+                        statementWrite.setNull(i, Types.VARCHAR);
+                    }
+                    else {
+                        statementWrite.setString(i, (String)field.val);
+                    }
+                    break;
+                case BYTEA:
+                    if (field.val == null) {
+                        statementWrite.setNull(i, Types.BINARY);
+                    }
+                    else {
+                        statementWrite.setBytes(i, (byte[])field.val);
+                    }
+                    break;
+                case TIMESTAMP:
+                    if (field.val == null) {
+                        statementWrite.setNull(i, Types.TIMESTAMP);
+                    }
+                    else {
+                        statementWrite.setTimestamp(i, (Timestamp)field.val);
+                    }
+                    break;
+                case DATE:
+                    if (field.val == null) {
+                        statementWrite.setNull(i, Types.DATE);
+                    }
+                    else {
+                        statementWrite.setDate(i, (Date)field.val);
+                    }
+                    break;
+                default:
+                    throw new IOException("The data tuple from JdbcResolver is corrupted");

```

Comment:
This message is not very actionable. I'd be more specific and mention that certain data type is not supported.

Reply:
This message is correct. If a certain data type is not supported, the exception must be thrown [in `JdbcResolver`](https://github.com/arenadata/incubator-hawq/blob/pxf_jdbc_writeAndFix/pxf/pxf-jdbc/src/main/java/org/apache/hawq/pxf/plugins/jdbc/JdbcResolver.java#L223).